zerver/lib/actions.py

```diff
@@ -5485,6 +5485,38 @@ class DeleteMessagesEvent(TypedDict, total=False):
     stream_id: int
 
 
+class UpdateMessagesEvent(TypedDict, total=False):
```

Comment:
Why total=False here?

Reply:
That is something I used as a temporary solution and need to fix(hence the current WIP status). Removing it causes error since `message_ids`, which is a compulary field is calculated much later in the code and the event TypedDict is created much before.

Reply:
@timabbott Something that I now remembered -- the `total=False` is necessary because we need to omit the Optional parameters completely rather than including them with `None`, since the API expects those keys to be absent rather than present with value `None`, and gives error in such cases. 
A downside of doing this is that we won't be able to check for missing parameters in the event.(I followed the example of `DeleteMessagesEvent` to use `total=False`)